config/generic_update.py

```diff
@@ -0,0 +1,332 @@
+    def __init__(self, patchorderer = PatchOrderer(), changeapplier = ChangeApplier(), configlocker = ConfigLocker(), configwrapper = ConfigWrapper()):
+        # NOTE: patch-applier receives a NoOpConfigLocker
+        self.patchapplier = PatchApplier(patchorder, changeapplier, NoOpConfigLocker(), configwrapper)
+        self.configwrapper = configwrapper
+        self.configlocker = configlocker
+
+    def replace(self, full_json, format, verbose, dry_run):
+        if format == "ConfigDB":
+            yang_as_json = convert_config_db_to_yang(full_json)
+        elif format == "YANG":
+            yang_as_json = full_json
+        else:
+            raise AttributeError(f"format argument value '{format}', the supported values are 'ConfigDB' or 'YANG'")
+
+        if verbose:
+            # TODO: Implement verbose logging
+            raise NotImplementedError
+
+        if dry_run:
+            # TODO: Implement dry-run
+            raise NotImplementedError
+
+        __acquire_config_lock()
+        if not(__validate(target_config)):
+            raise Exception(f"The given target config is not valid")
+
+        old_config = __get_current_config()
+        patch = __generate_patch(old_config, target_config)
+
+        __apply_patch(patch, verbose, dry_run)
+
+        new_config = __get_current_config()
+        if not(__verify_same_json(target_config, new_config)):
+            raise Exception(f"After applying patch to config, there is still some parts not updated")
+
+        __release_config_lock()
+
+    def __acquire_config_lock():
+        self.configlocker.acquire_lock()
+
+    def __release_config_lock():
+        self.configlocker.release_lock()
+
+    def __validate(target_config):
+        return self.configwrapper.validate_config(target_config)
+
+    def __get_current_config():
+        self.configwrapper.get_current_config()
+
+    def __generate_patch(current, target):
+        return jsonpatch.make_patch(current, target)
+
+    def __apply_patch(patch, verbose, dry_run):
+        self.patchapplier.apply(patch, verbose, dry_run)
+
+    def __verify_same_json(expected_json, actual_json):
+        return jsonpatch.make_patch(expected_json, actual_json) == []
+
+CHECKPOINTS_DIR = "/etc/sonic/checkpoints"
+CHECKPOINT_EXT = ".cp.json"
+class FileSystemRollbacker:
+    def __init__(self, patchorderer = PatchOrderer(), changeapplier = ChangeApplier(), configlocker = ConfigLocker(), configwrapper = ConfigWrapper()):
+        # NOTE: config-replacer receives a NoOpConfigLocker
+        self.configreplacer = PatchApplier(patchorder, changeapplier, NoOpConfigLocker(), configwrapper)
+        self.configlocker = configlocker
+
+    def rollback(self, checkpoint_name, verbose, dry_run):
+        if verbose:
+            # TODO: Implement verbose logging
+            raise NotImplementedError
+
+        if dry_run:
+            # TODO: Implement dry-run
+            raise NotImplementedError
+
+        __acquire_config_lock()
```

Comment:
>__ [](start = 8, length = 2)

Most function names are prefix with double underscore, which makes the file very busy.
Do you have a strong reason to hide them?